### PR TITLE
Added support for the new RHSM reporter's name.

### PR DESCRIPTION
**Pull request by roliveri**
# Overview

This PR is being created to address [ESSNTL-4621](https://issues.redhat.com/browse/ESSNTL-4621).

Currently, reports that come from the `rhsm-conduit` reporter don't require an identity specification, as they use the SYSTEM identity. Reports from the new `rhsm-system-profile-bridge` need to be handled in the same way.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by thearifismail — COMMENTED** [review]
Minor comments, else looks good.

For reminding users of the utility script, add *bridge* to the comment.
https://github.com/RedHatInsights/insights-host-inventory/blob/0fdb7dafab6462edc87856941bd2c2f1882c316d/utils/payloads.py#L581

File: tests/test_mq_service.py [hunk 1]
```diff
@@ -553,27 +555,28 @@ def test_add_host_rhsm_conduit_without_cn(event_datetime_mock, mq_create_or_upda
...
             account=SYSTEM_IDENTITY.get("account_number"),
-            reporter="rhsm-conduit",
+            reporter=rhsm_reporter,
             subscription_manager_id=sub_mangager_id,
         )
     else:
-        host = minimal_host(reporter="rhsm-conduit", subscription_manager_id=sub_mangager_id)
+        host = minimal_host(reporter=rhsm_reporter, subscription_manager_id=sub_mangager_id)
 
     key, event, headers = mq_create_or_update_host(host, platform_metadata=metadata_without_b64, return_all_data=True)
```
[the remark this answers is not in the document]
Comment (roliveri) [reply tests/test_mq_service.py:564]: Same as the previous comment. That input parameter was in the original code. Not sure it's appropriate to remove it as part of this change.

File: tests/test_mq_service.py [hunk 2]
```diff
@@ -553,27 +555,28 @@ def test_add_host_rhsm_conduit_without_cn(event_datetime_mock, mq_create_or_upda
...
 
     # owner_id equals subscription_manager_id with dashes
     assert event["host"]["system_profile"]["owner_id"] == "09152341-475c-4671-a376-df609374c349"
 
 
-def test_add_host_rhsm_conduit_owner_id(event_datetime_mock, mq_create_or_update_host):
+@pytest.mark.parametrize("rhsm_reporter", ("rhsm-conduit", "rhsm-system-profile-bridge"))
+def test_add_host_rhsm_conduit_owner_id(event_datetime_mock, mq_create_or_update_host, rhsm_reporter):
```
[the remark this answers is not in the document]
Comment (roliveri) [reply tests/test_mq_service.py:571]: Same as the previous comment. That input parameter was in the original code. Not sure it's appropriate to remove it as part of this change.

File: tests/test_mq_service.py [hunk 3]
```diff
@@ -535,13 +535,15 @@ def test_add_host_with_wrong_owner(mocker, event_datetime_mock, mq_create_or_upd
...
+    "with_account, rhsm_reporter",
+    [
+        (True, "rhsm-conduit"),
+        (False, "rhsm-conduit"),
+        (True, "rhsm-system-profile-bridge"),
+        (False, "rhsm-system-profile-bridge"),
+    ],
 )
-def test_add_host_rhsm_conduit_without_cn(event_datetime_mock, mq_create_or_update_host, with_account):
+def test_add_host_rhsm_conduit_without_cn(event_datetime_mock, mq_create_or_update_host, with_account, rhsm_reporter):
```
[the remark this answers is not in the document]
Comment (thearifismail) [reply tests/test_mq_service.py:546]: Removing the unused `event_datetime_mock` arg did not cause any problems, when I tested it locally.

**Comment by thearifismail** [comment]
LGTM'd because the issues were trivial.

File: tests/test_mq_service.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kruai) [reply tests/test_mq_service.py:546]: `event_datetime_mock` turns the datetimes for produced events into mocks for easier testing. This test doesn't appear to need it, but it also doesn't hurt. Great if removed, shouldn't hold up the PR if not

**Review by kruai — APPROVED** [review]

**Review by CodeHeeler — APPROVED** [review]
LGTM, thanks!

**Pull request merged**